trees/python/binary-tree.py

```diff
@@ -0,0 +1,66 @@
+from collections import deque
```

Comment:
Please rename the file to `algorithms/Python/trees/binary_tree.py` to march the changes made in #207